ducktape/tests/result_store.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2015 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.utils.util import ducktape_version
+
+import os
+import json
+import sqlite3
+import time
+
+
+def create_test_datum(result):
+
+    return {
+        "timestamp": time.time(),
+        "ducktape_version": ducktape_version(),
+        "session_id": result.session_context.session_id,
+        "description": result.description,
+        "test_id": result.test_context.test_id,
+        "module_name": result.test_context.module_name,
+        "cls_name": result.test_context.cls_name,
+        "function_name": result.test_context.function_name,
+        "injected_args": result.test_context.injected_args,
+        "run_time_sec": result.run_time,
+        "services": result.test_context.services,
+        "status": "pass" if result.success else "fail",
+        "error_msg": result.summary,
+        "data": result.data
```

Comment:
Ah, I think I see now. Instead of moving all the test run info to a sub-object, you've kept all the test-specific result data in a sub-object. That works too :)